src/mmrelay/migrate.py

```
@@ -0,0 +1,2588 @@
+    """
+    Finalize a staged move by renaming the staging path to the final destination.
+
+    Parameters:
+        staging_path (Path): The path where the unit was staged.
+        dest_path (Path): The final destination path.
+
+    Raises:
+        OSError: If validation or finalization fails.
+    """
+    if not staging_path.exists():
+        raise OSError(f"Staging path does not exist: {staging_path}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)